reconcile/utils/jira_client.py

```diff
@@ -48,6 +43,50 @@ class IssueType(BaseModel):
     statuses: list[str]
 
 
+class FieldOption(BaseModel):
+    """A standard buildin issue field option."""
+
+    name: str
+
+    def __eq__(self, value: Any) -> bool:
```

Comment:
why override eq for `FieldOption` and `CustomFieldOption`, and only `value` from `JiraBoardIssueFieldV1` used to compare with it?

Reply:
I want to compare the `JiraBoardIssueFieldV1.value` strings against either `FieldOption` or  `CustomFieldOption`, regardless of the `name` vs `value` attributes. The Jira API is stupid, it's `name` for standard jira fields, but `value` for custom ones :(  

```
{
  "security": {
    "name": "Employee"
  },
  "custom_attr": {
    "value": "whatever"
  }
}

Reply:
do we need to differentiate between `FieldOption` and `CustomFieldOption`, or just flat them into `str` to let `options` be `list[str]`?

Reply:
Yes, we need to differentiate between them. It's done in the [caching part](https://github.com/app-sre/qontract-reconcile/blob/878c0c86ba88338ee17734a689e72a7203349837/reconcile/jira_permissions_validator.py#L167). This will produce the dictionary mentioned above

Reply:
so in jira alert template, we need to keep the same format as server response, use `name` for standard fields, `value` for custom fields?

Reply:
Unfortunately, yes